paderbox/visualization/plot.py

```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
     ax.set_xlabel('frame index')
     ax.set_ylabel('decode')
     return ax
+
+
+@create_subplot
+def barh(y, width, height=0.8, left=None, *, ax=None, scale=1, **kwargs):
+    """
+    A wrapper around `plt.barh` that can also handle
+    `pb.array.interval.ArrayInterval`s.
+
+    Args:
+        y: The `y` coordinate(s). Can be a scalar (all bars are drawn at the
+            same y coordinate) or array-like of length `len(width)` (each
+            element of `width` has its own y coordinate)
+        width: Scalar, array-like, `pb.array.interval.ArrayInterval` or list of
+            `pb.array.interval.ArrayInterval`.
+        height:
+        left:
+        ax:
+        scale:
+        **kwargs:
+
+    """
+    import paderbox as pb
+    from paderbox.array.interval import ArrayInterval
+
+    def _convert_array_interval(array_interval):
+        """Converts an ArrayInterval into values for `left` and `width` that
+        can be passed to plt.hbar"""
+        left = [
+            interval[0] * scale
+            for interval in array_interval.normalized_intervals
+        ]
+        width = [
+            (interval[1] - interval[0]) * scale
+            for interval in array_interval.normalized_intervals
+        ]
+        return left, width
+
+    if isinstance(width, ArrayInterval):
+        assert left is None, left
+        left, width = _convert_array_interval(width)
+    elif isinstance(width, list) and isinstance(
+            width[0], ArrayInterval
+    ):
+        if not pb.utils.nested.nested_all(width, lambda x: isinstance(x, ArrayInterval)):
+            raise ValueError()
+        assert left is None, left
+        if not isinstance(y, Iterable):
+            y = [y] * len(width)
+        assert len(y) == len(width), (y, len(width))
+        left = []
+        new_width = []
+        new_y = []
+        for w, y_ in zip(width, y):
+            left_, width_ = _convert_array_interval(w)
+            left.extend(left_)
+            new_width.extend(width_)
+            new_y.extend([y_] * len(width_))
+        width = new_width
+        y = new_y
+    elif not isinstance(width, Iterable):
+        width = [width]
+    ax.barh(y, width, height, left, **kwargs)
+    return ax
+
+
+@create_subplot
+def axvspan(start, stop=None, ax=None, **kwargs):
```

Comment:
this code does not work with the default ax, so either we have to change the default or reset it to plt if ax=None

Reply:
It works with the default `ax`. It gets filled by the `create_subplot` decorator, as for all other functions in this file

Reply:
ah I didnt see the decorator you are right of course